client/components/Trimmer.tsx

```diff
@@ -0,0 +1,142 @@
+import { useEffect, useRef, useState } from "react";
+import { getPreviewUrl, Video } from "../api";
+import { ProjectState } from "./Editor";
+import { RangeSlider } from "./RangeSlider";
+
+type Clip = ProjectState["clips"][0];
```

Comment:
Move this and project state into types and define project state in terms of clip